fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/ContainmentIndexImpl.java

```diff
@@ -696,4 +712,38 @@ private Timestamp formatInstant(final Instant instant) {
         timestamp.setNanos(0);
         return timestamp;
     }
+
+    /**
+     * Private class to back a stream with a paged DB query.
+     *
+     * If this needs to be run in parallel we will have to override trySplit() and determine a good method to split on.
+     */
+    private class ContainmentIterator extends Spliterators.AbstractSpliterator<String> {
+        final Queue<String> children = new LinkedBlockingQueue<>();
```

Comment:
I think you might want a `ConcurrentLinkedQueue` here instead: https://www.baeldung.com/java-queue-linkedblocking-concurrentlinked

Incidentally, I think your original implementation of the parallelized reindex code would have worked if you'd backed the executor service with a `LinkedBlockingQueue`. I'm sorry I didn't think of it at the time. :(